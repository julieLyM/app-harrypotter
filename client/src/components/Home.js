import React, { Component } from 'react';
import { getHatSrv } from '../service/data';
import { HomeContainer, HomeTitle, HomeButton } from './styles/homeStyle';
import { houses } from '../images/griffon.png';
export default class Home extends Component {
  state = {
    hats: [],
    isLoading: false,
  };

  componentDidUpdate(prevProps, prevState) {
    if (this.state.hats === null) {
      this.onClicked();
    }
  }

  fetchHat = async () => {
    const data = await getHatSrv();
    this.setState({
      hats: data,
    });
  };

  onClicked = async () => {
    await this.fetchHat();
  };
  render() {
    const { hats } = this.state;
    return (
      <HomeContainer>
        <HomeTitle>Clic which houses are you in ?</HomeTitle>
        <img src={houses} alt="" style={{width:'100px', height:'100px'}}/>
        <HomeButton onClick={this.onClicked}>clic</HomeButton>
        {hats}
      </HomeContainer>
    );
  }
}
