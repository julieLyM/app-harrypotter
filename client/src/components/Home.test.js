import React from 'react';
import { shallow, simulate } from 'enzyme';
import Home from './Home';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

configure({ adapter: new Adapter() });

test('My Page renders correctly', () => {
  const tree = renderer.create(<Home />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Component should contain title', () => {
  const home = shallow(<Home />);
  expect(home.text()).toMatch(/houses/);
});

// test('simulates click events', () => {
//   const onButtonClick = jest.fn();
//   const button = shallow(<Home onClicked={onButtonClick} />);
//   button.find('button').simulate('click');
//   expect(onButtonClick).toHaveBeenCalledTimes(1);
// });
