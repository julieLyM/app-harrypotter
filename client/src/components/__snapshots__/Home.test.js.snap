// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Page renders correctly 1`] = `
<div
  className="sc-AxjAm jILPE"
>
  <p
    className="sc-AxirZ hYTMGg"
  >
    Clic which houses are you in ?
  </p>
  <button
    className="sc-AxiKw jBuzWk"
    onClick={[Function]}
  >
    clic
  </button>
</div>
`;
